File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    cents = get_cents()  # getting how many cents, by using our function
    coins = 0  # keeping track of number of coins

    # Calculate number of quarters
    while cents >= 25:
        cents = cents - 25
        coins = coins + 1  # we cant get coins++ like in C

    # Calculate number of dimes
    while cents >= 10:
        cents = cents - 10
        coins = coins + 1

    # Calculate number of nickels
    while cents >= 5:
        cents = cents - 5
        coins = coins + 1

    # Calculate number of pennies
    while cents >= 1:
        cents = cents - 1
        coins = coins + 1

    print(f"Total coins: {coins}")  # using string interpolation


# So i tried doing multiple functions like the cash in C language, but that was the only function i can abstract
# doing the division like in C does not work here like to obtain quarters cents / 25, you can't get here
# but i think  that option  was more easier than C version, i just needed to think a little more
def get_cents():
    while True:
        cents = get_float("Change owed ")
        if cents > 0:
            break
    cents = round(cents * 100)
    return cents

main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{

    // Ask how many cents the customer is owed

    int cents  = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    int next = quarters % 25;
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    next = dimes % 10;
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    next = nickels % 5;
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    next = pennies % 1;
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;



    // Print total number of coins to give the customer
    printf("Coins here %i\n", coins);
}

int get_cents()
{
    int cents;
    do
    {
        cents = get_int("How Many cents?\n");
    }
    while (cents < 0);

    return cents;
}

int calculate_quarters(int cents)
{
    int quarters = cents / 25; // The coin value
    return quarters;
}

int calculate_dimes(int cents)
{
    int dimes = cents / 10;
    return dimes;
}

int calculate_nickels(int cents)
{
    int nickels = cents / 5;
    return nickels;
}

int calculate_pennies(int cents)
{
    int pennies = cents / 1;
    return pennies;
}
